corehq/apps/hqadmin/views/users.py

```diff
@@ -520,19 +520,35 @@ class AppBuildTimingsView(TemplateView):
     def get_context_data(self, **kwargs):
         context = super(AppBuildTimingsView, self).get_context_data(**kwargs)
         app_id = self.request.GET.get('app_id')
+        request_user_id = self.request.couch_user._id
         if app_id:
             try:
                 app = Application.get(app_id)
             except ResourceNotFound:
                 raise Http404()
+            timing_context = self.get_timing_context(app, request_user_id)
             context.update({
                 'app': app,
-                'timing_data': self.get_timing_context(app).to_list(),
+                'timing_data': timing_context.to_list(),
             })
         return context
 
     @staticmethod
-    def get_timing_context(app):
+    def get_timing_context(app, request_user_id):
         with TimingContext(app.name) as context:
-            pass  # TODO build the app
+
+            with context('validate app'):
+                errors = app.validate_app()
+
+            assert not errors, errors
+
+            with context('make build'):
+                copy = app.make_build(
+                    comment="Generated automatically during profiling",
```

Comment:
I don't know that people will be generating enough of these for it to be an issue, but there's [an option](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/tasks.py#L33) to mark a build to be deleted automatically the next time a new build is made.

Reply:
oh nice, I'll do that

Reply:
oh wait nevermind - this doesn't save the new build, so I think that's not relevant